v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -47,16 +60,160 @@ async function getTasksAndHreEnvironment(
 }
 
 describe("main", function () {
-  let hre: HardhatRuntimeEnvironment;
-  let tasks: NewTaskDefinition[];
-  let subtasks: NewTaskDefinition[];
+  describe("parseInitialHardhatParameters", function () {
+    it("should set the all the hardhat initial global parameters", async function () {
+      // All the <value> and "task" should be ignored
+      const command =
+        "npx hardhat --help <value> --version --show-stack-traces task --config ./path-to-config <value>";
+
+      const cliArguments = command.split(" ").slice(2);
+      const usedCliArguments = new Array(cliArguments.length).fill(false);
+
+      const { configPath, showStackTraces, help, version } =
+        await parseInitialHardhatParameters(cliArguments, usedCliArguments);
+
+      assert.deepEqual(usedCliArguments, [
+        true,
+        false,
+        true,
+        true,
+        false,
+        true,
+        true,
+        false,
+      ]);
+      assert.equal(configPath, "./path-to-config");
+      assert.equal(showStackTraces, true);
+      assert.equal(help, true);
+      assert.equal(version, true);
+    });
+
+    it("should not set any hardhat initial global parameters except the path to the config", async function () {
```

Comment:
I think this title is incorrect

Reply:
[Done in this commit](https://github.com/NomicFoundation/hardhat/pull/5334/commits/f45d1f1e4442fad9871c2ced4fe4bc53899dbdca)